queries/ocaml/hightlights.scm

```diff
@@ -0,0 +1,146 @@
+(module_name) @structure
+
+[
+  (module_type_name)
+  (type_constructor)
+  (type_variable)
+  (constructor_name)
+  (tag)
+] @type
+(type_binding (type_constructor) @type)
+
+(exception_definition (constructor_declaration (constructor_name) @exception))
+(application_expression (value_path (value_name) @_raise)
+                        (#eq? @_raise "raise")
+                        (constructor_path) @exception)
+(application_expression (value_path (value_name) @_raise)
+                        (#eq? @_raise "raise")
+                        (parenthesized_expression
+                        (application_expression (constructor_path) @exception)))
+(infix_expression (value_path (value_name) @_raise)
+                  (#eq? @_raise "raise")
+                  (application_expression (constructor_path) @exception))
+
+(attribute_id) @property
+(field_name) @field
+
+(let_binding
+  (value_name) @function
+  (parameter))
+(let_binding
+  (value_name) @function
+  (function_expression))
+
+(application_expression . (value_path (value_name) @function))
+(application_expression . (value_path (value_path (value_name) @function)))
+(application_expression . (value_path (value_path (value_path (value_name) @function))))
+(value_specification (value_name) @function (function_type))
+(infix_expression (value_path (value_name) @function)
+                  (infix_operator) @operator
+                  (#eq? @operator "@@"))
+(infix_expression (value_path (value_path (value_name) @function))
+                  (infix_operator) @operator
+                  (#eq? @operator "@@"))
+(infix_expression (value_path (value_path (value_path (value_name) @function)))
+                  (infix_operator) @operator
+                  (#eq? @operator "@@"))
+(infix_expression (infix_operator) @operator
+                  (#eq? @operator "|>")
+                  (value_path (value_name) @function))
+(infix_expression (infix_operator) @operator
+                  (#eq? @operator "|>")
+                  (value_path (value_path (value_name) @function)))
+(infix_expression (infix_operator) @operator
+                  (#eq? @operator "|>")
+                  (value_path (value_path (value_path (value_name) @function))))
+
```

Comment:
This is not ideal. But I don't see a better solution for now since it would require combining two queries. A predicate for this situation would need to be really specialized. Can we extend the has-ancenstor to accept a Lua function as a predicate for the parent? Or a predicate just for that situation?

Reply:
I've raised an issue to the ocaml syntax, I think the tree structure can be changed to avoid this uglyness. https://github.com/tree-sitter/tree-sitter-ocaml/issues/40